### ISSUE TITLE
Sub-nav for categories

### ISSUE THREAD

**Issue by CiaranChris** [briggySmalls/skin-deep-server#114]
We may need to add and remove a few categories. 

I have a call with Edna today at 4 to run through the categories and seeing if there are any additional ones that may need to be added. 

Example of one that is a bit hard to place - https://skindeepmag.tk/articles/the-moon-is-a-meme/

Also, just a note to remove any categories that aren't going to be in use, magazine, radical interruptions, imagining 2043 , i assume these just ported over from the old site. Will suggest to Edna and Anu about using tags. What do you think?

**Comment by briggySmalls** [comment]
Yeah, the old ones are ported. No problem removing them.

Adding new categories is fine, only concern is whether they will fit on menu.

Tags vs. categories: The way tags work in wordpress is that it is _really_ easy to create new ones. You type in a text box and it will find a matching one if it exists, otherwise it creates a new one.

So tags are probably not something you'd put in the navigation menu (because they are always being created / more dynamic), but more just display as a 'cloud' on the archive card.

If that still fits the needs then we can explore how we want to display tags

**Comment by briggySmalls** [comment]
\*sigh* that sub-menu was not easy to make...

I think we did discuss what happens when more categories are inevitably added.

Obviously the downside to _always_ collapsing the menu by default is that people don't see the list of categories unless they go hunting for it. The plus side is you can have as many/few as you want and it doesn't risk overflowing the menu. 🤷‍♂️

**Comment by briggySmalls** [comment]
p.s. I don't understand what you mean by 'tags'. Can you create a separate issue to discuss this as it doesn't seem related to categories?

**Comment by CiaranChris** [comment]
TAGS: haha, I'm the source of much of this confusion. I thought tags (I guess their called meta tags) were still a thing that search engines used to help better rank a story and website. I don't mean for them to interfere with the categories and the way this works on the site.

_Meta Keywords Tags
At the beginning of the optimization race, meta keywords used to be small snippets of text only visible in the code, that were supposed to tell the search engines what topics the page relates to.

Naturally, over the years the tag turned into a breeding ground for spamming and stuffing, instead of honestly optimizing the content.

Now, it’s a well-known fact that Google ignores meta keywords completely – they neither impact the rankings, nor would cause a penalty if you stuff it up.

Bottom line: meta keywords are pretty much obsolete and not worth wasting too much of your time on.

Following best SEO practices, you should:

Only put relevant and reasonable keywords in it, or ignore the field the same way Google does._

https://www.searchenginejournal.com/important-tags-seo/156440/

**Comment by CiaranChris** [comment]
SUB-MENU: Dang ok. I would rather the sub-menu was visible on a bigger screen where possible, it helps the user, myself included, know what topics are covered and navigation :') 

_Obviously the downside to always collapsing the menu by default is that people don't see the list of categories unless they go hunting for it._

Completely agree with you on the collapsable menu, no hunting.

**Comment by briggySmalls** [comment]
So now I really don't know what you are suggesting.

> SUB-MENU: Dang ok. I would rather the sub-menu was visible on a bigger screen where possible, it helps the user, myself included, know what topics are covered and navigation :')

This is what we have already. A sub-menu visible on bigger screens.

> This is my concern also. There are a couple of alternatives we could try for V2. Jacobin and Dazed use what looks like a mobile menu for navigation which could align with the menu you are currently putting together.

This statement contradicts the first, here you suggest a mobile (read collapsable) menu.

The mobile menu will still have sub-items. It just won't require the work I did before.

**Comment by CiaranChris** [comment]
Sorry, I seem to be confusing you. 

I'm saying I like and prefer that the menu and sub-menus are displayed, rather than a collapsable menu. 

The suggestion for collapsible across both desktop and mobile was just if it worked out to be easier and more consistent, design wise, to apply this style of menu to both. However this was only a suggestions and not something I would like to implement across both.

Sorry sorry sam

**Comment by briggySmalls** [comment]
Sorry. I was grouchy yesterday. not your fault.

Incidentally, in addition to apple.com, I've noticed that https://www.moneysavingexpert.com/ and github.com itself use that horizontally scrollable menu approach.

Notice how moneysavingexpert has a little popup prompt suggesting you scroll horizontally

**Comment by CiaranChris** [comment]
Not at all. Sometimes it happens. 

moneysavingexpert is really nice, that pop up 👍🏾. The colours underneath are a nice touch, similar to the guardian. Great to see it could be an option 😁

**Comment by briggySmalls** [comment]
Cool. I'll look into it. I suppose the skin deep logo would stay where it is on desktop, but like github and moneysavingexpert it would appear above the categories when on mobile

**Comment by briggySmalls** [comment]
Apple use arrows to indicate ability to scroll:

<img width="387" alt="screen shot 2018-12-01 at 11 32 49" src="https://user-images.githubusercontent.com/11054133/49327740-e4d8a600-f55c-11e8-94d5-fcc6a8f5b9d1.png">

They also _don't_ display a horizontal scrollbar, whereas the other two (it should be noted that this might not be possible to do on all browsers - e.g. Microsoft Edge might always show a scrollbar).

**Comment by briggySmalls** [comment]
I'm going to start with a scrollbar as this is the easiest thing to implement first-off

**Comment by briggySmalls** [comment]
I don't really know what to do with the "Articles, Events, Products" top-line part of the navbar on mobile, if the categories will be horizontally scrollable... We can't have _two_ horizontally scrolling navs...

**Comment by briggySmalls** [comment]
@CiaranChris, I have pretty much finished this feature now. However I'm not sure what to do about _affordance_ on mobile (when the horizontal scrolling is enabled). Are there any visual cues for horizontal scrolling that you favour. Some options are:

- Popup hand a la moneysavingexpert. This is quite clear, but I don't like things that disappear after a time duration. How do you know the user saw it? I also don't know how hard it is to show it only once during a user's session, so I've avoided it thus far. But if it's your favourite I can investigate.
- [Dots, arrows, illusion of continuity](https://www.nngroup.com/articles/mobile-carousels/). The suggestions in the article are really for carousels not navigation menus, but there are some good tips there. It suggests to me maybe separators between categories? Or even better maybe border boxes around each category. This way you can see when there are more categories that overflow off the page. Similarly underlines that reach the full width of the link box, a la moneysavingexpert's coloured lines.

**Comment by briggySmalls** [comment]
phew 😅 that took some work.

Can you have a look at the new menu and let me know.

Particularly look out of how I've given the sub-menu a border on mobile to ensure there is always indication of further items off the edge of the page. This styling could change, but I really think it is important to make it clear the horizontal scrolling is possible.

**Comment by CiaranChris** [comment]
Sam, you the man!!! this is complete 🔥

Loving the new sub-menu. Agreed. 

I'm guessing the appearance scroll bar must be OS dependant. No scroll bar on iOS (my iPhone). Do you have a screen shot, depending on how ugly it be, if we can't beautify with the design then lets get rid.

**Comment by briggySmalls** [comment]
Ah maybe so (OS A dependent).

This is what it looks like on mine 
![screenshot_20181206-234022](https://user-images.githubusercontent.com/11054133/49618465-9db14180-f9b0-11e8-9aeb-c2808a402758.png)

Note: in order to get the margin either side of the sub menu, unfortunately i can't find a way to confine the scroll bar within the sub menu. You can see it extends to the left side till it is flush with the edge of the screen instead. I couldn't find a way to remove it.

So i guess i vote to remove it. I just ran out of time last night.

**Comment by CiaranChris** [comment]
https://www.youtube.com/watch?v=0cgbZqR2AGI

second this vote. It's got to go. 😅

**Comment by briggySmalls** [comment]
I struggled with hiding the scrollbar. I'll make a separate issue for it (v2) and close this. I'll also make a separate issue for additional categories, as this discussion has drifted.
